### PR TITLE
Adds Feedback Intake Form "Toast"

### LINKED ISSUE #1205: Need a survey / feedback feature
Add Survey to collect info about user backgrounds and what is and isn't working on the tracker for them

Proposal to add to:
- Landing page
- Download and Methodology
- About Us

flexible on how it presents - pop up? or toast notification (Ben idea). Can we add logic - only presents when user scrolls to end of page? cookie settings to only present to returning users?

### PR DESCRIPTION
**Pull request by benhammondmusic**
TODO (some might be post-launch)
- [x] finalize approved wording/options
- [x] fix sentence casing on questions
- [ ] consider adding a "hide this" button especially for tracker page
- [x] allow user to minimize feedback toast by clicking outside the box

Adds to:
- Both Data / Methodology Tabs (scroll to bottom)
- Explore Data (scroll to bottom, should hide when user scrolls back up?)
- Project Tab (scrolled to bottom)
- Contact Us (always on)

Feedback box only appears for returning users (as defined by the presence of the onboarding cookie). This means that the following users would never be shown the feedback bubble: 
- a first time user of the site
- a returning user who has cookies disabled
- a returning user who has never been on the tracker tab
- a user who never scrolls to the very bottom of any of the pages (other than the about us page)

TO DEMO
To view as user, scroll to the bottom of one of these pages; a small icon will appear in the bottom right corner. Hover to view message, and click to expand popper with feedback form. Submit feedback

To view all feedback as team member: log in at https://feeder.sh/dashboard (assuming you've made an account and been granted access to the project)

closes #1205

### CODE REVIEW

**Comment by benhammondmusic** [comment]
https://user-images.githubusercontent.com/41567007/138367477-b22aa1e0-e0f3-458f-a8df-e8dbdc553ef5.mov

MOBILE

**Comment by benhammondmusic** [comment]
PROCESS
- Cloned original feedback component, edits are made on my clone
- publish my clone to name scoped npm package
- import my version to HET rather than the original

UPDATES
- allows vertical menu for readability of more options
- more control of styling / layout / wording
- allows unlimited changes to the number and types of questions

TO DO
- [ ] see how these altered data are received / processed by the backend dashboard and accommodate

https://user-images.githubusercontent.com/41567007/141197916-29affd27-4111-4fa9-9619-847fc1b754bb.mov

**Comment by benhammondmusic** [comment]
Current proposed question(s) (should be approved):

## What brought you to the Health Equity Tracker?
- General/Professional Interest
- Covid Data, Covid-19 Vaccination Data
- Social/Political Determinants (SPDOH) Data
- Mental Health Data
- Academic Interest
- Professional/Advocacy
- Legal/Policy Support
- Other (freetext? with character limit?)

## What field of work/study are you in? (should this be combined with the previous question??)
- Community Engagement
- Non-Profit
- Legal/Political
- Medical/Clinical
- Academic/Teaching
- Academic/Student
- Professional
- Other, N/A

## Did you get what you needed today?
Free text

**Comment by benhammondmusic** [comment]
Adjusted so any added questions/responses get built in to the message, allowing easy viewing on the backend dashboard

Screen Shot 2021-11-10 at 4 48 08 PM

**Comment by benhammondmusic** [comment]
Images at different device widths. I still think we need to reduce the number of options if possible; would love to avoid the "scroll within a scroll" that occurs on mobile

Screen Shot 2021-11-10 at 9 17 03 PM
Screen Shot 2021-11-10 at 9 17 21 PM
Screen Shot 2021-11-10 at 9 17 31 PM
Screen Shot 2021-11-10 at 9 17 41 PM
Screen Shot 2021-11-10 at 9 17 56 PM

**Comment by benhammondmusic** [comment]
- [ ] finalize approved wording/options
- [ ] fix sentence casing on questions
- [ ] consider adding a "hide this" button especially for tracker page

**Comment by benhammondmusic** [comment]
https://user-images.githubusercontent.com/41567007/143940456-ee87deb5-b024-4b3a-8aaa-61eedb9ae8c1.mov

**Review by joshzarrabi — COMMENTED** [review]
Was this file checked in on purpose? frontend/package-lock.json~8816012f2891886accba66908b4b4fff6e424dda

File: frontend/src/pages/ui/FeedbackBox.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import React, { useState } from "react";
+// @ts-ignore
+import Feedback from "@benhammondmusic/feeder-react-feedback";
+import "@benhammondmusic/feeder-react-feedback/dist/feeder-react-feedback.css"; // import stylesheet
+
+import { useBottomScrollListener } from "react-bottom-scroll-listener";
+import { useCookies } from "react-cookie";
+
+const FEEDBACK_ID = "6171cc2965b82c00045239dc"; // view collected feedback at feeder.sh/dashboard
```
Comment (joshzarrabi) [inline frontend/src/pages/ui/FeedbackBox.tsx:9]: I think we should pass this in as secret.

**Review by joshzarrabi — CHANGES_REQUESTED** [review]
Can we make the box get smaller again if you click outside of it back to the tracker? I think thats a little bit of a better UI.

**Comment by benhammondmusic** [comment]
> Was this file checked in on purpose? `frontend/package-lock.json~8816012f2891886accba66908b4b4fff6e424dda`

not sure where that came from.... i deleted it and it's still working so can't have been too important lol

**Comment by benhammondmusic** [comment]
> Can we make the box get smaller again if you click outside of it back to the tracker? I think thats a little bit of a better UI.

good call. i made this a task above. We could launch without, I'll take a look at what's involved; shouldn't be too difficult

File: frontend/src/pages/ui/FeedbackBox.tsx (hunk 1 above)
Comment (benhammondmusic) [reply frontend/src/pages/ui/FeedbackBox.tsx:9]: https://github.com/SatcherInstitute/health-equity-tracker/blob/main/.github/workflows/deployToTest.yml#L86

**Comment by benhammondmusic** [comment]
Close the feedback box if user clicks away

https://user-images.githubusercontent.com/41567007/144290734-35482787-d395-418a-985f-29596b3370a2.mov

File: frontend/src/pages/ui/FeedbackBox.tsx (hunk 1 above)
Comment (benhammondmusic) [reply frontend/src/pages/ui/FeedbackBox.tsx]: Currently I set it up to read from process.env or this ID. once we have production working reading github secrets, we can create a new dashboard account at feeder.sh and use that new secret id

**Comment by benhammondmusic** [comment]
> > Can we make the box get smaller again if you click outside of it back to the tracker? I think thats a little bit of a better UI.
> 
> good call. i made this a task above. We could launch without, I'll take a look at what's involved; shouldn't be too difficult

got this working! even better, I made it a reusable generic hook for any component we might want to add this "click away to dismiss" function

File: frontend/src/pages/ui/FeedbackBox.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+// @ts-ignore
+import Feedback from "@benhammondmusic/feeder-react-feedback";
+import "@benhammondmusic/feeder-react-feedback/dist/feeder-react-feedback.css"; // import stylesheet
+import { useBottomScrollListener } from "react-bottom-scroll-listener";
+import { useCookies } from "react-cookie";
+import useClickAway from "../../utils/useClickAway";
+import sass from "../../styles/variables.module.scss";
+
+const TEST_FEEDER_ID = "6171cc2965b82c00045239dc"; // this is a test account; we can create a new ID for our real dashboard account once secrets are working
+const FEEDBACK_ID = process.env.REACT_APP_FEEDBACK_ID || TEST_FEEDER_ID; // view collected feedback at feeder.sh/dashboard
```
Comment (benhammondmusic) [inline frontend/src/pages/ui/FeedbackBox.tsx:11]: Replace this once we have our production and deploy test secrets working

**Comment by benhammondmusic** [comment]
Did you mean to close this PR before I merged it?

File: frontend/src/pages/ui/FeedbackBox.tsx (hunk 2 above)
Comment (joshzarrabi) [reply frontend/src/pages/ui/FeedbackBox.tsx:11]: Nice

**Review by joshzarrabi — APPROVED** [review]

**Comment by joshzarrabi** [comment]
Looks good to me

**Pull request merged**